arq/worker.py

```diff
@@ -509,6 +515,10 @@ async def job_failed(exc: BaseException) -> None:
             'score': score,
         }
         ctx = {**self.ctx, **job_ctx}
+
+        if self.on_job_start:
```

Comment:
should it be wrapped in try .. except?

Reply:
i chose not to do this because on_startup and on_shutdown also don't wrap a try catch, if you think it is needed, i can add it.

it certainly adds a lot more complexity though because you need to check all the different types of exceptions